tool/tsh/common/tsh.go

```diff
@@ -2162,6 +2160,34 @@ func onLogout(cf *CLIConf) error {
 			}
 		}
 
+		// If any of the profiles have SAML SLO (single logout) enabled, fetch all clusters clients.
+		var clusters []*clusterClient
+		for _, profile := range profiles {
+			if profile.SAMLSingleLogoutEnabled {
+				clusters, err = getClusterClients(cf, "")
+				if err != nil {
+					return trace.Wrap(err)
+				}
+				break
+			}
+		}
+
+		for _, cluster := range clusters {
+			if cluster.profile.SAMLSingleLogoutEnabled {
+				SAMLSingleLogoutURL, err := tc.GetSAMLSingleLogoutURL(cf.Context, cluster.cluster, cluster.profile)
+				if err != nil {
+					fmt.Printf("We were unable to log you out of your SAML identity provider: %v", err)
+				}
+				// Initiate SAML SLO (single logout) if there is a relevant SLO URL.
+				if SAMLSingleLogoutURL != "" {
+					err := tc.SAMLSingleLogout(cf.Context, SAMLSingleLogoutURL)
+					if err != nil {
+						fmt.Printf("We were unable to log you out of your SAML identity provider: %v", err)
+					}
+				}
+			}
+		}
+
```

Comment:
How about something like this instead (but with logging and some better UX)?
```suggestion
		forEachProfileParallel(cf, func(ctx context.Context, tc *client.TeleportClient, profile *client.ProfileStatus) error {
			if !profile.SAMLSingleLogoutEnabled {
				return nil
			}
			clt, err := tc.ConnectToCluster(ctx)
			if err != nil {
				return trace.Wrap(err)
			}
			defer clt.Close()
			sloURL, err := tc.GetSAMLSingleLogoutURL(ctx, clt, profile)
			if err != nil {
				return trace.Wrap(err)
			}
			if sloURL == "" {
				return nil
			}
			return trace.Wrap(tc.SAMLSingleLogout(ctx, sloURL))
		})

```